CRM/Financial/Form/FrontEndPaymentFormTrait.php

```diff
@@ -43,25 +43,39 @@ trait CRM_Financial_Form_FrontEndPaymentFormTrait {
    * CRM_Invoicing_Utils class with a potential end goal of moving this handling to an extension.
    *
    * @param $tplLineItems
-   *
-   * @return array
    */
-  protected function alterLineItemsForTemplate($tplLineItems) {
-    $getTaxDetails = FALSE;
+  protected function alterLineItemsForTemplate(&$tplLineItems) {
+    if (!CRM_Invoicing_Utils::isInvoicingEnabled()) {
+      return;
+    }
+    // @todo this should really be the first time we are determining
+    // the tax rates - we can calculate them from the financial_type_id
+    // & amount here so we didn't need a deeper function to semi-get
+    // them but not be able to 'format them right' because they are
+    // potentially being used for 'something else'.
+    // @todo invoicing code - please feel the hate. Also move this 'hook-like-bit'
+    // to the CRM_Invoicing_Utils class.
     foreach ($tplLineItems as $key => $value) {
       foreach ($value as $k => $v) {
         if (isset($v['tax_rate']) && $v['tax_rate'] != '') {
-          $getTaxDetails = TRUE;
+          $this->assign('getTaxDetails', TRUE);
+          $this->assign('taxTerm', CRM_Invoicing_Utils::getTaxTerm());
```

Comment:
We are calling these lines for every lineitem when they only need assigning once.  I can see why, just not sure if that's really a good idea.

Reply:
Not sure I agree. It should be possible for different line items to have different taxes configured, leading to a different taxTerm being displayed for different line items.

Reply:
@JoeMurray Agree.  But `getTaxDetails` and `taxTerm` are being assigned globally so they'll just overwrite the same var each time the loop is run.  `tax_rate` *is* per lineitem, but `taxTerm` is a global setting under "Tax and Invoicing" and CiviCRM doesn't currently support multiple tax terms (eg. Sales Tax, IVA, VAT).

Reply:
@mattwire yeah they only need to be assigned once but it seemed really 'cheap' & to make the code more readable. It doesn't change functionality at all

Reply:
Ok, still not completely comfortable as I worry it has the potential to introduce performance issues later (eg. if we made getTaxTerm() do something wierd and wonderful).  If you add a one line comment `// These only need assigning once, but code is more readable with them here` then I'll merge